nvflare/apis/workspace.py

```diff
@@ -135,11 +168,33 @@ def get_site_custom_dir(self) -> str:
     def get_startup_kit_dir(self) -> str:
         return os.path.join(self.root_dir, WorkspaceConstants.STARTUP_FOLDER_NAME)
 
-    def get_audit_file_path(self) -> str:
-        return os.path.join(self.root_dir, WorkspaceConstants.AUDIT_LOG)
+    def get_audit_root(self, job_id=None) -> str:
+        if self.audit_root:
+            return self._get_site_root_dir(self.audit_root, job_id)
+        elif job_id:
+            return self.get_run_dir(job_id)
+        else:
+            return self.root_dir
 
-    def get_log_file_path(self) -> str:
-        return os.path.join(self.root_dir, WorkspaceConstants.LOG_FILE_NAME)
+    def get_audit_file_path(self, job_id=None) -> str:
+        return os.path.join(self.get_audit_root(job_id), WorkspaceConstants.AUDIT_LOG)
+
+    def _get_site_root_dir(self, root, job_id=None):
+        if job_id:
+            site_root_dir = os.path.join(root, self.site_name, job_id)
+            if not os.path.exists(site_root_dir):
```

Comment:
This pattern is used several times. Maybe we should create a convenience function.

Reply:
This method _get_site_root_dir is the general pattern that is reused in multiple places.